packages/charts/src/components/legend/legend_icon.tsx

```diff
@@ -9,36 +9,43 @@
 import React from 'react';
 
 import { ShapeRendererFn } from '../../chart_types/xy_chart/renderer/shapes_paths';
-import { Color } from '../../utils/common';
-import { PointStyle } from '../../utils/themes/theme';
-import { Icon } from '../icons/icon';
+import { Color, isNil } from '../../utils/common';
+import { PointShape, PointStyle } from '../../utils/themes/theme';
 
 interface LegendIconProps {
   pointStyle?: PointStyle;
   color: Color;
 }
 
-const MARKER_SIZE = 16;
+const MARKER_SIZE = 8;
+
+// to limit size, set a min and max
+const getRatio = (radius: number | undefined, strokeWidth: number) => {
+  const adjustedRadius = isNil(radius) ? strokeWidth : ((MARKER_SIZE / 2) * strokeWidth) / (radius * 2);
+  if (1.5 > adjustedRadius) return 1.5;
+  return Math.min(adjustedRadius, (radius ?? strokeWidth) * 2);
```

Comment:
I think we could just set a static min of `1px` and max of the `MARKER_SIZE / 2`

```suggestion
  const adjustedRadius = isNil(radius) ? strokeWidth : ((MARKER_SIZE / 2) * strokeWidth) / (radius * 2);

  return Math.max(Math.min(adjustedRadius, MARKER_SIZE / 2), 1);
```

Reply:
Woah! Cool 80c4cea